services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -305,58 +299,86 @@ private void recordBasicMetrics() {
+        totalStats.recordReadComputeLatency(readComputeLatency, isAssembledMultiChunkLargeValue());
+        serverHttpRequestStats.recordReadComputeLatency(readComputeLatency, isAssembledMultiChunkLargeValue());
       }
       if (readComputeDeserializationLatency >= 0) {
-        currentStats.recordReadComputeDeserializationLatency(
-            storeName,
+        totalStats.recordReadComputeDeserializationLatency(
+            readComputeDeserializationLatency,
+            isAssembledMultiChunkLargeValue());
+        serverHttpRequestStats.recordReadComputeDeserializationLatency(
             readComputeDeserializationLatency,
             isAssembledMultiChunkLargeValue());
       }
       if (readComputeSerializationLatency >= 0) {
-        currentStats.recordReadComputeSerializationLatency(
-            storeName,
-            readComputeSerializationLatency,
-            isAssembledMultiChunkLargeValue());
+        totalStats
+            .recordReadComputeSerializationLatency(readComputeSerializationLatency, isAssembledMultiChunkLargeValue());
+        serverHttpRequestStats
+            .recordReadComputeSerializationLatency(readComputeSerializationLatency, isAssembledMultiChunkLargeValue());
       }
       if (dotProductCount > 0) {
-        currentStats.recordDotProductCount(storeName, dotProductCount);
+        totalStats.recordDotProductCount(dotProductCount);
+        serverHttpRequestStats.recordDotProductCount(dotProductCount);
       }
       if (cosineSimilarityCount > 0) {
-        currentStats.recordCosineSimilarityCount(storeName, cosineSimilarityCount);
+        totalStats.recordCosineSimilarityCount(cosineSimilarityCount);
+        serverHttpRequestStats.recordCosineSimilarityCount(cosineSimilarityCount);
       }
       if (hadamardProductCount > 0) {
-        currentStats.recordHadamardProductCount(storeName, hadamardProductCount);
+        totalStats.recordHadamardProduct(hadamardProductCount);
+        serverHttpRequestStats.recordHadamardProduct(hadamardProductCount);
       }
       if (countOperatorCount > 0) {
-        currentStats.recordCountOperatorCount(storeName, countOperatorCount);
+        totalStats.recordCountOperator(countOperatorCount);
+        serverHttpRequestStats.recordCountOperator(countOperatorCount);
       }
       if (isRequestTerminatedEarly) {
-        currentStats.recordEarlyTerminatedEarlyRequest(storeName);
+        totalStats.recordEarlyTerminatedEarlyRequest();
+        serverHttpRequestStats.recordEarlyTerminatedEarlyRequest();
+      }
+      if (keySizeList != null) {
+        for (int i = 0; i < keySizeList.size(); i++) {
+          totalStats.recordKeySizeInByte(keySizeList.getInt(i));
+          serverHttpRequestStats.recordKeySizeInByte(keySizeList.getInt(i));
+        }
+      }
+      if (valueSizeList != null) {
+        for (int i = 0; i < valueSizeList.size(); i++) {
+          totalStats.recordValueSizeInByte(valueSizeList.getInt(i));
```

Comment:
Since the latest code change is doing set by index on this list, I wonder what happens here if the value was not found in the storage engine. Is it zero? Would it make sense to record zero if that's the case? Should we record -1 instead and have the code here skip recording the value size metric since the value wasn't found?